core/src/main/java/io/questdb/griffin/engine/table/LatestByAllFilteredRecordCursor.java

```diff
@@ -53,17 +53,22 @@ public LatestByAllFilteredRecordCursor(
         this.map = map;
         this.recordSink = recordSink;
         this.filter = filter;
+        this.isOpen = true;
     }
 
     @Override
     public void close() {
-        filter.close();
-        super.close();
+        if (isOpen) {
+            isOpen = false;
+            filter.close();
+            map.close();
+            super.close();
+        }
     }
 
     @Override
     protected void buildTreeMap(SqlExecutionContext executionContext) throws SqlException {
-        map.clear();
+        map.inflate();
```

Comment:
Here as well, please do not inflate something which was just created in the constructor